openassessment/workflow_batch_update_api.py

```diff
@@ -0,0 +1,629 @@
+"""
+Provides functionality to batch update ORA workflows for different scopes
+"""
+
+import logging
+import time
+import datetime
+from django.utils import timezone
+from celery import shared_task
+
+from opaque_keys.edx.keys import UsageKey
+from openassessment.runtime_imports.functions import modulestore
+from openassessment.assessment.models import PeerWorkflow
+from openassessment.workflow import api
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_all_blocked_submissions_task(self):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_all_blocked_submissions()
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_course_task(self, course_id):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_course(course_id)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_ora_block_task(self, item_id):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_ora_block(item_id)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=300,
+             retry_jitter=True)
+# pylint: disable=unused-argument
+def update_workflow_for_submission_task(self, submission_uuid, assessment_requirements=None, course_settings=None):
+    """
+    Async task wrapper
+    """
+    return update_workflow_for_submission(submission_uuid, assessment_requirements, course_settings)
+
+
+def logging_decorator(func):
```

Comment:
Nit: update for more descriptive title.

```suggestion
def log_result_and_time(func):
```

Reply:
It is more than that so didn't want to define the purpose more finely.

Reply:
Minimally should be something like `log_task_info`, even if you keep it that vague. Also having "decorator" after a decorator reads like "logging decorator decorator" and is redundant.

Or would require a docstring update, you can't say "does what it says" if it doesn't actually "say what it does".

Reply:
Also the logs should probably include the scope / function name. It does appear to have a description, but probably want something more machine-searchable.